### DISCUSSION TITLE
Edge middleware rewrite doesn't work

### DISCUSSION THREAD

**Discussion by cosbgn** [vercel/community#4807] [Bug Report]
### Page to Investigate

https://github.com/cosbgn/edgenu/blob/main/middleware.js

### Steps to Reproduce

This middleware here: https://github.com/cosbgn/edgenu/blob/main/middleware.js

should return the content of https://edgenu.vercel.app/api/backend if the route https://edgenu.vercel.app/foo is visited. Instead it returns a 404. I've verified that the middleware works correctly with the logs, it's just that rewrite doesn't work.

**Comment by wsuchy** [comment]
I've got a similar issue. The following middleware works fine when running locally but gets ignored on Vercel:
```ts
export async function middleware(request: NextRequest) {
  console.log("MIDDLEWARE", request.nextUrl.href)
  console.log("MIDDLEWARE", request.cookies.getAll())
  const now  = new Date()

  const resp = NextResponse.next()

  resp.cookies.set('foo', now.toISOString(), {
    httpOnly: true,
  })

  resp.headers.set('x-foo', now.toISOString())
  return resp
  ```
